lib/cache-activator.js

```diff
@@ -0,0 +1,53 @@
+const hasher = require('node-object-hash');
+const {startUpdater, cache} = require('./cache-manager');
+const CacheItem = require('./cache-item');
+const hashSha1 = hasher({alg: 'sha1'});
+
+let originalPromise = null;
+
+function augment(pp, fnName) {
+  const fetch = pp[fnName];
+  pp[fnName] = async function(...args) {
+    const sha1 = hashSha1.hash(args);
+    let item = cache[sha1];
+    if (!item) {
+      item = new CacheItem(sha1, pp, fetch, args);
+      await item.refresh();
+      cache[sha1] = item;
```

Comment:
I believe that we should not have a reference to item, but rather a copy of it.  Otherwise we are going to inadvertently have our cached item be modified after being put in cache if the caller operates on it.

So perhaps we need to make a deep copy
```js
cache[sha1] = JSON.parse(JSON.stringify(item));
```

Reply:
This makes sense